report_generator_3.py

```diff
@@ -0,0 +1,18 @@
+import subprocess
+
+def generate_report(report_title, content_file):
+    """Generates a PDF report with a user-supplied title."""
+    
+
+    command = f"pandoc {content_file} -o report.pdf --metadata title='{report_title}'"
```

Comment:
Critical Command Injection

The `report_title` variable, which directly originates from unsanitized user input (`user_title`), is used to construct a shell command executed via `subprocess.run` with `shell=True`. This allows for command injection, where an attacker can execute arbitrary commands on the system by providing a malicious `report_title`.

```suggestion
    command = ["pandoc", content_file, "-o", "report.pdf", "--metadata", f"title={report_title}"]
    result = subprocess.run(command, capture_output=True, text=True)
```